cmd/local-gadget/top/top.go

```diff
@@ -0,0 +1,160 @@
+	commontop.TopGadget[Stats]
+
+	commonFlags        *utils.CommonFlags
+	stats              []*Stats
+	createAndRunTracer func(*ebpf.Map, gadgets.DataEnricher, func(*top.Event[Stats])) (trace.Tracer, error)
+}
+
+func addCommonTopFlags[Stats any](
+	command *cobra.Command,
+	commonTopFlags *commontop.CommonTopFlags,
+	commonFlags *utils.CommonFlags,
+	colMap columns.ColumnMap[Stats],
+	sortBySliceDefault []string,
+) {
+	command.Flags().IntVarP(&commonTopFlags.MaxRows, "max-rows", "m", top.MaxRowsDefault, "Maximum rows to print")
+	validCols, _ := sort.FilterSortableColumns(colMap, colMap.GetColumnNames())
+	command.Flags().StringVarP(
+		&commonTopFlags.SortBy, "sort",
+		"",
+		strings.Join(sortBySliceDefault, ","), fmt.Sprintf("Sort by columns. Join multiple columns with ','. Prefix a column with '-' to sort in descending order. Available columns: (%s)", strings.Join(validCols, ", ")))
+	utils.AddCommonFlags(command, commonFlags)
+}
+
+// Run runs a TopGadget and prints the output after parsing it using the
+// TopParser's methods.
+func (g *TopLocalGadget[Stats]) Run(args []string) error {
+	localGadgetManager, err := localgadgetmanager.NewManager(g.commonFlags.RuntimeConfigs)
+	if err != nil {
+		return commonutils.WrapInErrManagerInit(err)
+	}
+	defer localGadgetManager.Close()
+
+	// TODO: Improve filtering, see further details in
+	// https://github.com/inspektor-gadget/inspektor-gadget/issues/644.
+	containerSelector := containercollection.ContainerSelector{
+		Name: g.commonFlags.Containername,
+	}
+
+	// Create mount namespace map to filter by containers
+	mountnsmap, err := localGadgetManager.CreateMountNsMap(containerSelector)
+	if err != nil {
+		return commonutils.WrapInErrManagerCreateMountNsMap(err)
+	}
+	defer localGadgetManager.RemoveMountNsMap()
+
+	if len(args) == 1 {
+		g.CommonTopFlags.OutputInterval, err = strconv.Atoi(args[0])
+		if err != nil {
+			return commonutils.WrapInErrInvalidArg("<interval>", fmt.Errorf("%q is not a valid value", args[0]))
+		}
+	} else {
+		g.CommonTopFlags.OutputInterval = top.IntervalDefault
+	}
+
+	sortByColumns := strings.Split(g.CommonTopFlags.SortBy, ",")
+	_, invalidCols := sort.FilterSortableColumns(g.ColMap, sortByColumns)
+
+	if len(invalidCols) > 0 {
+		return commonutils.WrapInErrInvalidArg("--sort", fmt.Errorf("invalid columns to sort by: %q", strings.Join(invalidCols, ",")))
+	}
+	g.CommonTopFlags.ParsedSortBy = sortByColumns
+
+	g.StartPrintLoop()
+
+	// Define a callback to be called each time there is an event.
+	eventCallback := func(event *top.Event[Stats]) {
+		g.stats = event.Stats
+	}
+
+	gadgetTracer, err := g.createAndRunTracer(mountnsmap, &localGadgetManager.ContainerCollection, eventCallback)
+	if err != nil {
+		return commonutils.WrapInErrGadgetTracerCreateAndRun(err)
+	}
+	defer gadgetTracer.Stop()
+
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, syscall.SIGINT, syscall.SIGTERM)
+	<-stop
+
+	return nil
+}
+
+func (g *TopLocalGadget[Stats]) PrintStats() {
```

Comment:
Seeing that this function is not called by `TopLocalGadget.Run`, I just realized local-gadget doesn't have the timeout flag 🤦🏽. I will add it.